src/base/init/init.c

```diff
@@ -404,15 +404,25 @@ void low_mem_init(void)
   }
 
   if (config.ext_mem) {
-    /* LOWMEM_SIZE + HMASIZE == base
-     * Use dpmi_rsv_low as ext_mem. */
-    register_hardware_ram_virtual('X', LOWMEM_SIZE + HMASIZE, EXTMEM_SIZE,
-	    MEM_BASE32(LOWMEM_SIZE + HMASIZE), LOWMEM_SIZE + HMASIZE);
+    /* LOWMEM_SIZE + HMASIZE == base */
     memcheck_addtype('X', "EXT MEM");
     memcheck_reserve('X', LOWMEM_SIZE + HMASIZE, EXTMEM_SIZE);
     x_printf("Ext.Mem of size 0x%x at %#x\n", EXTMEM_SIZE, result + LOWMEM_SIZE + HMASIZE);
   }
 
+  /* create non-identity mapping up to dpmi_base */
+  phys_rsv = config.dpmi_base - (LOWMEM_SIZE + HMASIZE);
+  ptr = smalloc(&main_pool, phys_rsv);
```

Comment:
This I would also limit then:
```
phys_rsv = _min(config.dpmi_base - (LOWMEM_SIZE + HMASIZE), (LOWMEM_SIZE + HMASIZE + EXTMEM_SIZE + config.xms_size*1024);
```

and this ptr is the one I mentioned that doesn't conflict with `&mem_base[dpmi_base]`.